fleet/agent_group_service.go

```diff
@@ -235,6 +237,37 @@ func (svc fleetService) RemoveAgentGroup(ctx context.Context, token, groupId str
 		svc.logger.Error("removing agents from group channel", zap.Error(errors.Wrap(ErrMaintainAgentGroupChannels, err)))
 	}
 
+	group, err := svc.agentGroupRepository.RetrieveByID(ctx, groupId, ownerID)
+	if err != nil {
+		return err
+	}
+
+	var lastRead uint64 = 1
+	for {
+		connectedAgents, err := svc.mfsdk.ThingsByChannel(token, group.MFChannelID, lastRead, limitThingsByChannel, false)
+		if err != nil {
+			return err
+		}
+
+		for _, agent := range connectedAgents.Things {
+			err := svc.mfsdk.DisconnectThing(agent.ID, group.MFChannelID, token)
+			if err != nil {
+				return errors.Wrap(errors.New("error while disconnecting agents"), err)
```

Comment:
i don't think we should return an error here, because it will leave other agents still connected and orphaned. let's log the error but continue through the list